projects/transformers/tests/sparse_bert_model_test.py

```diff
@@ -0,0 +1,129 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import sys
+import tempfile
+import unittest
+from os.path import expanduser
+
+from transformers import (
+    CONFIG_MAPPING,
+    MODEL_FOR_MASKED_LM_MAPPING,
+    MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING,
+    AutoModelForMaskedLM,
+    AutoModelForSequenceClassification,
+)
+
+sys.path.insert(0, expanduser("~/nta/nupic.research/projects/transformers")) # noqa
+from models import (
+    StaticSparseEncoderBertConfig,
+    StaticSparseEncoderBertForMaskedLM,
+    StaticSparseEncoderBertForSequenceClassification,
+)
+from nupic.torch.modules import SparseWeights
+# noqa
+
+
+class SparseBertModelTest(unittest.TestCase):
+    """
+    Test the ability to load and save StaticSparseEncoderBert models and associated
+    StaticSparseEncoderBertConfig.
+    """
+
+    def setUp(self):
+        self.config = CONFIG_MAPPING["static_sparse_encoder_bert"](
```

Comment:
Same comment as above, this network is actually quite big and disproportional in terms of number of weights to attention and to non-attention layers. It has zero impact on test except for speed, but if we change in experiments we might as well reflect those changes here.

Reply:
I changed it, mostly for speed of the test.